### PR TITLE
[Federation][init-10a] Make the federation API server and controller manager image names configurable.

### PR DESCRIPTION
**Pull request by madhusudancs**
This enables testing non-release images.

Please review only the last commit here. This is based on PR #35959 which will be reviewed independently.

Design Doc: PR #34484

cc @kubernetes/sig-cluster-federation @nikhiljindal

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36045)

### CODE REVIEW

**Comment by quinton-hoole** [comment]
Thanks @madhusudancs.  Only nits, then LGTM.  Feel free to apply the label once addressed to your satisfaction.

**Comment by madhusudancs** [comment]
@quinton-hoole I don't see the comments (nits). It is not published?

**Comment by quinton-hoole** [comment]
@madhusudancs Yes, sorry, I forgot to press the "publish" button :-(

File: federation/pkg/kubefed/init/init.go [hunk 1]
```diff
@@ -109,8 +110,11 @@ func NewCmdInit(cmdOut io.Writer, config util.AdminConfig) *cobra.Command {
 		},
 	}
 
+	defaultImage := fmt.Sprintf("%s:%s", hyperkubeImageName, version.Get())
```
Comment (madhusudancs) [inline federation/pkg/kubefed/init/init.go]: @quinton-hoole Yeah, that's the most ideal default behavior. But there is no good/stable way to automatically detect the latest version right now. There is no API or anything as such to determine the latest version on the client machines and no technique is future proof. Given that, I think this is a reasonable default.

File: federation/pkg/kubefed/init/init.go [hunk 2]
```diff
@@ -109,8 +110,11 @@ func NewCmdInit(cmdOut io.Writer, config util.AdminConfig) *cobra.Command {
...
+
 	util.AddSubcommandFlags(cmd)
 	cmd.Flags().String("dns-zone-name", "", "DNS suffix for this federation. Federated Service DNS names are published with this suffix.")
+	cmd.Flags().String("image", defaultImage, "Image to use for federation API server and controller manager binaries.")
```
Comment (madhusudancs) [inline federation/pkg/kubefed/init/init.go]: @quinton-hoole `defaultImage` is printed in the help text. I think that illustrates a valid value. If I were to add an example, I would just repeat that.

**Pull request merged**